internal/resolution/variable_sources/bundles_and_dependencies/bundles_and_dependencies.go

```diff
@@ -0,0 +1,166 @@
+package bundles_and_dependencies
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/blang/semver/v4"
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/constraint"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+
+	olmentity "github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/required_package"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/utils/predicates"
+	entitysort "github.com/operator-framework/operator-controller/internal/resolution/variable_sources/utils/sort"
+)
+
+type BundleVariable struct {
+	*input.SimpleVariable
+	bundleEntity *olmentity.BundleEntity
+	dependencies []*olmentity.BundleEntity
+}
+
+func (b *BundleVariable) BundleEntity() *olmentity.BundleEntity {
+	return b.bundleEntity
+}
+
+func (b *BundleVariable) Dependencies() []*olmentity.BundleEntity {
+	return b.dependencies
+}
+
+func NewBundleVariable(bundleEntity *olmentity.BundleEntity, dependencyBundleEntities []*olmentity.BundleEntity) *BundleVariable {
+	var dependencyIDs []deppy.Identifier
+	for _, bundle := range dependencyBundleEntities {
+		dependencyIDs = append(dependencyIDs, bundle.ID)
+	}
+	var constraints []deppy.Constraint
+	if len(dependencyIDs) > 0 {
+		constraints = append(constraints, constraint.Dependency(dependencyIDs...))
+	}
+	return &BundleVariable{
+		SimpleVariable: input.NewSimpleVariable(bundleEntity.ID, constraints...),
+		bundleEntity:   bundleEntity,
+		dependencies:   dependencyBundleEntities,
```

Comment:
For my understanding:
- SimpleVariables seems to take each dependency bundleID for purposes of resolution.
- dependencies are the actual entities, which we seem to store here for quick reference later rather than retrieving the entity from the source, is that correct?

Reply:
`SimpleVariable` is the simplest implementation of `deppy.Variable` possible. It's just a convenience. Basically, it just has an id and a list of constraints and you can `Add`  constraints. It will contain everything the resolver needs to convert this to some boolean expression.

`dependencies` are the actual bundle dependencies - exactly so we don't need to go back to the backing store to get them later.